support search gan model. 1.add pixel_unshuffle support &2.enable fill_constant calc offline on arm and opencl & 3.enable reshape_calc_offline_pass on arm and opencl

xiebaiyuan opened the pull request:
### PR devices
Arm

### PR types
Op Support & Performance optimization

### PR changes
OP | PASS | Kernels |

### Description
1.add pixel_unshuffle support
2.enable fill_constant calc offline on arm and opencl
3.enable reshape_calc_offline_pass on arm and opencl

hong19860320 commented on lite/operators/pixel_unshuffle_op.cc:
```diff
@@ -0,0 +1,87 @@
+  const auto x_dims = param_.x->dims();
+  const auto downscale_factor = param_.downscale_factor;
+
+  // 检查输入张量的维度数量是否为4
```
注释是否统一成英文？

hong19860320 commented on lite/kernels/arm/pixel_unshuffle_compute.cc:
```diff
@@ -0,0 +1,66 @@
+  int out_height = param.output->dims()[2];
+  int out_width = param.output->dims()[3];
+
+  for (int b = 0; b < batch_size; ++b) {
```
如果是 naive 实现，建议放到 host 目录下

hong19860320 commented on lite/operators/pixel_unshuffle_op.h:
```diff
@@ -0,0 +1,57 @@
+class PixelUnShuffleOpLite : public OpLite {
+ public:
+  PixelUnShuffleOpLite() {}
+  explicit PixelUnShuffleOpLite(const std::string &op_type) : OpLite(op_type) {}
```
是否能增加该算子的单测？

hong19860320 reviewed (dismissed):
LGTM

xiebaiyuan commented:
看起来流水线是各种clone资源失败。 我这里好像没有权限rerun
@hong19860320

xiebaiyuan committed:
support search gan model.
1. add pixel_unshuffle support
2. enable fill_constant calc offline on arm and opencl
3. enable reshape_calc_offline_pass on arm and opencl
4. use chinese comments
5. add test for new kernel.
test=develop

xiebaiyuan committed:
support search gan model.
1. add pixel_unshuffle support
2. enable fill_constant calc offline on arm and opencl
3. enable reshape_calc_offline_pass on arm and opencl
4. use chinese comments
5. add test for new kernel.
test=develop

xiebaiyuan committed:
support search gan model.
1. add pixel_unshuffle support
2. enable fill_constant calc offline on arm and opencl
3. enable reshape_calc_offline_pass on arm and opencl
4. use chinese comments
5. add test for new kernel.
6. fix metal pre-commit
test=develop
